bundles/org.openhab.binding.anel/src/test/java/org/openhab/binding/anel/internal/AnelUdpConnectorTest.java

```diff
@@ -0,0 +1,185 @@
+
+        if (!connector.isConnected()) {
+            connector.connect(receivedMessages::offer, false);
+        }
+    }
+
+    @Test
+    public void connectionTest() throws Exception {
+        final String response = sendAndReceiveSingle(IAnelConstants.BROADCAST_DISCOVERY_MSG);
+        /*
+         * Expected example response:
+         * "NET-PwrCtrl:NET-CONTROL    :192.168.0.244:255.255.255.0:192.168.0.1:0.5.163.21.4.71:Nr. 1,0:Nr. 2,1:Nr. 3,0:Nr. 4,0:Nr. 5,0:Nr. 6,0:Nr. 7,0:Nr. 8,0:248:80:NET-PWRCTRL_04.6:H:xor:"
+         */
+        assertThat(response, startsWith(IAnelConstants.STATUS_RESPONSE_PREFIX + IAnelConstants.STATUS_SEPARATOR));
+    }
+
+    @Test
+    public void toggleSwitch1() throws Exception {
+        toggleSwitch(1);
+    }
+
+    @Test
+    public void toggleSwitch2() throws Exception {
+        toggleSwitch(2);
+    }
+
+    @Test
+    public void toggleSwitch3() throws Exception {
+        toggleSwitch(3);
+    }
+
+    @Test
+    public void toggleSwitch4() throws Exception {
+        toggleSwitch(4);
+    }
+
+    @Test
+    public void toggleSwitch5() throws Exception {
+        toggleSwitch(5);
+    }
+
+    @Test
+    public void toggleSwitch6() throws Exception {
+        toggleSwitch(6);
+    }
+
+    @Test
+    public void toggleSwitch7() throws Exception {
+        toggleSwitch(7);
+    }
+
+    @Test
+    public void toggleSwitch8() throws Exception {
+        toggleSwitch(8);
+    }
+
+    private void toggleSwitch(int switchNr) throws Exception {
+        assertThat(switchNr, allOf(greaterThan(0), lessThan(9)));
+        final int index = 5 + switchNr;
+
+        // get state of switch 1
+        final String status = sendAndReceiveSingle(IAnelConstants.BROADCAST_DISCOVERY_MSG);
+        final String[] segments = status.split(IAnelConstants.STATUS_SEPARATOR);
+        assertThat(segments[5 + switchNr], anyOf(endsWith(",1"), endsWith(",0")));
+        final boolean switch1state = segments[index].endsWith(",1");
+
+        // toggle state of switch 1
+        final String auth = AnelAuthentication.getUserPasswordString(USER, PASSWORD, AuthMethod.of(status));
+        final String command = "Sw_" + (switch1state ? "off" : "on") + String.valueOf(switchNr) + auth;
+        final String status2 = sendAndReceiveSingle(command);
+
+        // assert new state of switch 1
+        assertThat(status2.trim(), not(endsWith(":Err")));
+        final String[] segments2 = status2.split(IAnelConstants.STATUS_SEPARATOR);
+        final String expectedState = segments2[index].substring(0, segments2[index].length() - 1)
+                + (switch1state ? "0" : "1");
+        assertThat(segments2[index], equalTo(expectedState));
+    }
+
+    @Test
+    public void withoutCredentials() throws Exception {
+        final String status2 = sendAndReceiveSingle("Sw_on1");
+        assertThat(status2.trim(), endsWith(":NoPass:Err"));
+        Thread.sleep(3100); // locked for 3 seconds
```

Comment:
See above. Don't use sleep in tests. Please check all.

Reply:
Again, it will never run on CI. So I assume it should be fine.